From 24eb11f2d9ba13fbaa2b53508f88c3704ef896e6 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Thu, 1 Apr 2021 13:23:23 -0400
Subject: [PATCH 1/3] Reduce amount of debug info injected in extension modules

sysconfig.get_config_var('CFLAGS') has '-g', which defaults to level 2

Py310 builds on linux-aarch64 require >5GB with level 2, which result
in the docker build containers getting killed on azure, where we have
only 7GB of memory. With -g1, the memory requirement decreases to <4GB
---
 mypyc/build.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mypyc/build.py b/mypyc/build.py
index 088e481fc..27bad19ba 100644
--- a/mypyc/build.py
+++ b/mypyc/build.py
@@ -508,7 +508,7 @@ def mypycify(
         cflags += [
             '-O{}'.format(opt_level), '-Werror', '-Wno-unused-function', '-Wno-unused-label',
             '-Wno-unreachable-code', '-Wno-unused-variable',
-            '-Wno-unused-command-line-argument', '-Wno-unknown-warning-option',
+            '-Wno-unused-command-line-argument', '-Wno-unknown-warning-option', '-g1',
         ]
         if 'gcc' in compiler.compiler[0]:
             # This flag is needed for gcc but does not exist on clang.
-- 
2.33.1

